### PR TITLE
6220 holo testing

### PR DESCRIPTION
**Pull request by SJJCoding**
Pull request for 6220 holo testing, which includes the BaseOpMode, BaseTeleOp, and TeleOpTest methods. 

this code has identical hardwaremap names as the 6220 tele op branch but the drive method includes both joystick averaging for the last 5 recorded joystick inputs and stick dead zones for improved precision in controlling the robot. 

Additionally, there is a constants.js file which contains some constants useful for auto.

### CODE REVIEW

File: Team6220_PowerPlay/src/main/java/org/firstinspires/ftc/team6220_PowerPlay/ResourceClasses/Constants.java
Comment (tmicro) [inline Team6220_PowerPlay/src/main/java/org/firstinspires/ftc/team6220_PowerPlay/ResourceClasses/Constants.java:5]: You should derive the wheel circumference and inches per tick from the diameter.  That way, you only need to change the one parameter - diameter if you change to another size wheel.

File: Team6220_PowerPlay/src/main/java/org/firstinspires/ftc/team6220_PowerPlay/ResourceClasses/Constants.java
Comment (tmicro) [inline Team6220_PowerPlay/src/main/java/org/firstinspires/ftc/team6220_PowerPlay/ResourceClasses/Constants.java:6]: You should consider replacing this magic number 537.6 with an expression for ticks per revolution, based on the motor's gear ratio.  That way, the magic number changes if you change to motors with a different gear ratio.

File: Team6220_PowerPlay/src/main/java/org/firstinspires/ftc/team6220_PowerPlay/BaseTeleOp.java
Comment (tmicro) [inline Team6220_PowerPlay/src/main/java/org/firstinspires/ftc/team6220_PowerPlay/BaseTeleOp.java:9]: This method needs a design document.  Would you please write what this code tries to do, and how it does that?

File: .idea/.gitignore
Comment (tmicro) [inline .idea/.gitignore:1]: Please don't merge this file.  Remove it.

File: Team417_PowerPlay/src/main/java/org/firstinspires/ftc/team417_PowerPlay/TeleOpCompetition.java
Comment (tmicro) [inline Team417_PowerPlay/src/main/java/org/firstinspires/ftc/team417_PowerPlay/TeleOpCompetition.java:1]: Please don't merge this file.  You're not allowed to change another team's code.

File: Team6220_PowerPlay/src/main/java/org/firstinspires/ftc/team6220_PowerPlay/BaseTeleOp.java
Comment (tmicro) [inline Team6220_PowerPlay/src/main/java/org/firstinspires/ftc/team6220_PowerPlay/BaseTeleOp.java:1]: This file BaseTeleOp is the same one that Gavin already has a pull request out for.  You successfully tested it with him at the shop yesterday.  Please wait for his pull request to complete before merging this file.

**Review by tmicro — CHANGES_REQUESTED** [review]
Please wait until Gavin's pull request completes and make the suggested changes before merging this pull request.  Thank you.

File: Team417_PowerPlay/src/main/java/org/firstinspires/ftc/team417_PowerPlay/TeleOpCompetition.java
Comment (SJJCoding) [reply Team417_PowerPlay/src/main/java/org/firstinspires/ftc/team417_PowerPlay/TeleOpCompetition.java]: Not sure how this happened, but it will be changed

File: Team6220_PowerPlay/src/main/java/org/firstinspires/ftc/team6220_PowerPlay/BaseTeleOp.java
Comment (SJJCoding) [reply Team6220_PowerPlay/src/main/java/org/firstinspires/ftc/team6220_PowerPlay/BaseTeleOp.java:1]: Yes, this pull request will only be merged after coordinating with Gavin

File: Team417_PowerPlay/src/main/java/org/firstinspires/ftc/team417_PowerPlay/TeleOpCompetition.java
Comment (SJJCoding) [reply Team417_PowerPlay/src/main/java/org/firstinspires/ftc/team417_PowerPlay/TeleOpCompetition.java]: This has been resolved by the latest push

File: Team6220_PowerPlay/src/main/java/org/firstinspires/ftc/team6220_PowerPlay/ResourceClasses/Constants.java
Comment (tmicro) [inline Team6220_PowerPlay/src/main/java/org/firstinspires/ftc/team6220_PowerPlay/ResourceClasses/Constants.java:6]: For TicksPerRev = 14.147 * 2 * 19, where did 14.147 come from?  The expression doesn't seem correct.

File: Team6220_PowerPlay/src/main/java/org/firstinspires/ftc/team6220_PowerPlay/ResourceClasses/Constants.java
Comment (tmicro) [inline Team6220_PowerPlay/src/main/java/org/firstinspires/ftc/team6220_PowerPlay/ResourceClasses/Constants.java:4]: The wheel diameter 3.73 doesn't seem correct.

**Review by tmicro — CHANGES_REQUESTED** [review]

File: Team6220_PowerPlay/src/main/java/org/firstinspires/ftc/team6220_PowerPlay/BaseTeleOp.java
Comment (malavika1) [reply Team6220_PowerPlay/src/main/java/org/firstinspires/ftc/team6220_PowerPlay/BaseTeleOp.java:54]: Also what is so special about this number 20? Please make it a constant since you're using it in multiple places and we don't know what it means. https://en.wikipedia.org/wiki/Magic_number_(programming)

File: Team6220_PowerPlay/src/main/java/org/firstinspires/ftc/team6220_PowerPlay/ResourceClasses/Constants.java
Comment (malavika1) [reply Team6220_PowerPlay/src/main/java/org/firstinspires/ftc/team6220_PowerPlay/ResourceClasses/Constants.java:6]: Also where did the 2 and 19 come from?

**Pull request merged**
